pkg/placement/placement.go

```diff
@@ -234,12 +236,12 @@ func (p *Service) ReportDaprStatus(stream placementv1pb.Placement_ReportDaprStat
 
 	clientID, err := p.validateClient(stream)
 	if err != nil {
-		return err
+		return apiErrors.PlacementServiceInternalError(fmt.Sprintf("Internal Error: %s", codes.Internal))
 	}
 
 	firstMessage, err := p.receiveAndValidateFirstMessage(stream, clientID)
 	if err != nil {
-		return err
+		return apiErrors.PlacementServiceUnAuthenticated(fmt.Sprintf("Failed to get client ID from context with code: %s", codes.Unauthenticated))
```

Comment:
`p.receiveAndValidateFirstMessage` can return `codes.PermissionDenied` or `codes.Internal`  status codes with a specific error message, but here you're overriding them and replacing it with `codes.Unauthenticated`.